packages/apps/sale/src/hooks/use-variant-selection.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2024 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {useEffect, useState, useMemo} from 'react';
+import {fetchProductByIdApi} from '../api/';
+
+export const useVariantSelection = (product, productVariantConfig) => {
+  const [selectedVariants, setSelectedVariants] = useState<{
+    productVariantValue1?: any;
+    productVariantValue2?: any;
+    productVariantValue3?: any;
+    productVariantValue4?: any;
+    productVariantValue5?: any;
+  }>({});
+
+  const [alertVisible, setAlertVisible] = useState(false);
+
+  const variantConfig = useMemo(() => {
+    return productVariantConfig || null;
+  }, [productVariantConfig]);
+
+  const handleVariantSelection = () => {
+    setAlertVisible(true);
+  };
+
+  useEffect(() => {
+    if (product?.variantProduct != null && productVariantConfig) {
+      fetchProductByIdApi({productId: product?.variantProduct?.id})
+        .then(res => {
+          const variantData = res.data.data[0]?.productVariant;
+          if (variantData) {
+            setSelectedVariants({
+              productVariantValue1:
+                variantData.productVariantValue1 ?? undefined,
+              productVariantValue2:
+                variantData.productVariantValue2 ?? undefined,
+              productVariantValue3:
+                variantData.productVariantValue3 ?? undefined,
+              productVariantValue4:
+                variantData.productVariantValue4 ?? undefined,
+              productVariantValue5:
+                variantData.productVariantValue5 ?? undefined,
```

Comment:
To avoid code duplication you can map it, like:
`["productVariantValue"+number]: variantData["productVariantValue"+number] ?? undefined`